apps/examples/rtc/rtc_main.c

```diff
@@ -154,6 +161,228 @@ static void *rtc_test(void *arg)
 	return NULL;
 }
 
+#ifdef CONFIG_RTC_ALARM
+/****************************************************************************
+ * Name: alarm_handler
+ ****************************************************************************/
+
+static void alarm_handler(int signo, FAR siginfo_t *info, FAR void *ucontext)
+{
+	int ret;
+	int fd;
+	int secs;
+	int almndx;
+	FAR struct rtc_time alarm_recieve_time;
+
+	fd = open(CONFIG_EXAMPLES_RTC_ALARM_DEVPATH, O_WRONLY);
+	if (fd < 0) {
+		fprintf(stderr, "ERROR: Failed to open %s: %d\n", CONFIG_EXAMPLES_RTC_ALARM_DEVPATH, errno);
+		return;
+	}
+	almndx = info->si_value.sival_int;
+	if (almndx >= 0 && almndx < CONFIG_RTC_NALARMS) {
+		ret = ioctl(fd, RTC_RD_TIME, (unsigned long)&alarm_recieve_time);
+		if (ret < 0) {
+			fprintf(stderr, "ERROR: RTC_RD_TIME ioctl failed: %d\n", errno);
+			close(fd);
+			return;
+		}
+		close(fd);
+		secs = (alarm_recieve_time.tm_min - g_alarm_time[almndx].tm_min) * 60 + (alarm_recieve_time.tm_sec - g_alarm_time[almndx].tm_sec);
+		printf("Alarm %d received after %d seconds\n", almndx, secs);
+		g_alarm_received[almndx] = false;
+	}
+}
+
+/****************************************************************************
+ * Name: show_usage
+ ****************************************************************************/
+
+static void show_usage(FAR const char *progname)
+{
+	fprintf(stderr, "\nUsage:  rtc\n");
+	fprintf(stderr, "\tor: rtc stop\n");
+	fprintf(stderr, "\t\tStart, or Stop rtc accuracy test\n");
+	fprintf(stderr, "\tor %s alarm [-r seconds] [-a minutes seconds] [-c]\n", progname);
+	fprintf(stderr, "\t\tset, or cancel rtc alarm\n");
+	fprintf(stderr, "Where:\n");
+	fprintf(stderr, "\t-a <minutes> <seconds> \n");
+	fprintf(stderr, "\t\tset alarm by absolute time in minutes and seconds (default alarm id: 0)\n");
+	fprintf(stderr, "\t-r <seconds> \n");
+	fprintf(stderr, "\t\tset alarm by relative time. The number of seconds until the alarm expires (default alarm id: 0).\n");
```

Comment:
We dont need to mention anything about alarm id. Let us assume that we will always use alarm id 0 and at a given time a single alarm will be set.